gadgets/trace_exec/program.bpf.c

```diff
@@ -262,8 +267,13 @@ int BPF_KPROBE(security_bprm_check, struct linux_binprm *bprm)
 
 	f_path = BPF_CORE_READ(bprm, file, f_path);
 	file = get_path_str(&f_path);
+	struct inode *inode = BPF_CORE_READ(f_path.dentry, d_inode);
+	dev_no = BPF_CORE_READ(inode, i_sb, s_dev);
+	inode_no = BPF_CORE_READ(inode, i_ino);
 	bpf_probe_read_kernel_str(event->file, MAX_STRING_SIZE, file);
-
+	event->dev_major = (dev_no >> 20) & 0xfff;
+	event->dev_minor = dev_no & 0xfffff;      
```

Comment:
Instead of using the magic numbers, we could copy the macro and indicate where it comes from:

```c
// Macros from https://github.com/torvalds/linux/blob/v6.12/include/linux/kdev_t.h#L7-L12
#define MINORBITS       20
#define MINORMASK       ((1U << MINORBITS) - 1)
#define MAJOR(dev)      ((unsigned int) ((dev) >> MINORBITS))
#define MINOR(dev)      ((unsigned int) ((dev) & MINORMASK))
#define MKDEV(ma,mi)    (((ma) << MINORBITS) | (mi))
```